### PR TITLE
Fix create roles race condition

### PR DESCRIPTION
**Pull request by DarksightKellar**
Contract deployed on Sepolia for quick testing: `0xB1ba8caa9a65f4Bf53a5D1A1bD5514dFc9bB7E35`

There's a separate PR to merge in deploy artifacts on all networks

### CODE REVIEW

File: deployments/sepolia/DecentHats_0_1_0.json
Comment (adamgall) [inline deployments/sepolia/DecentHats_0_1_0.json:1]: Please remove the deployment artifacts from this PR, or, let's get this deployed on _all_ of the networks. I'd like contract deployments (on all chains) to happen "all at once" in the context of a PR.

File: deployments/sepolia/solcInputs/868499c9bf8e01dcdf8b61017b8261aa.json
Comment (adamgall) [inline deployments/sepolia/solcInputs/868499c9bf8e01dcdf8b61017b8261aa.json:1]: See previous comment

File: contracts/DecentHats_0_1_0.sol [hunk 1]
```diff
@@ -195,6 +195,29 @@ contract DecentHats_0_1_0 {
...
+            hatsProtocol,
+            adminHatId,
+            hat,
+            topHatAccount,
+            registry,
+            hatsAccountImplementation,
+            salt
+        );
+
+        hatsProtocol.transferHat(topHatId, address(this), msg.sender);
```
Comment (adamgall) [inline contracts/DecentHats_0_1_0.sol:232]: Can you please leave some natspec comments on this function, explaining how it's independent of the rest of the code in this contract, and why it exists, and the reasoning behind its implementation?

**Review by adamgall — CHANGES_REQUESTED** [review]

**Review by adamgall — APPROVED** [review]

**Review by mudrila — APPROVED** [review]
Noice!

**Review by dacebt — APPROVED** [review]

**Pull request merged**